File: core/src/main/java/forpleuvoir/dhwu1a/core/common/data/GroupData.java
package forpleuvoir.dhwu1a.core.common.data;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.user.Group;
import forpleuvoir.dhwu1a.core.user.base.Permission;

import static forpleuvoir.dhwu1a.core.common.ApiKey.NAME;
import static forpleuvoir.dhwu1a.core.common.ApiKey.PERMISSION;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.common.data
 * <p>#class_name GroupData
 * <p>#create_time 2021/7/3 21:58
 */
public class GroupData extends UserData<Group> {
    /**
     * 群名称
     */
    @SerializedName(NAME)
    public final String name;
    /**
     * Bot在群里的权限
     */
    @SerializedName(PERMISSION)
    public final Permission permission;

    public GroupData(Long id, String name, Permission permission) {
        super(id);
        this.name = name;
        this.permission = permission;
    }

    @Override
    public Group getUser() {
        return bot.getGroup(id);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/common/data/GroupData.kt
package forpleuvoir.dhwu1a.core.common.data

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.NAME
import forpleuvoir.dhwu1a.core.common.PERMISSION
import forpleuvoir.dhwu1a.core.user.Group
import forpleuvoir.dhwu1a.core.user.base.Permission

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.common.data
 *
 * #class_name GroupData
 *
 * #create_time 2021/7/3 21:58
 */
class GroupData(
    id: Long?,
    /**
     * 群名称
     */
    @field:SerializedName(NAME) var name: String,
    /**
     * Bot在群里的权限
     */
    @field:SerializedName(PERMISSION) var permission: Permission
) : UserData<Group>(
    id!!
) {
    override val user: Group
        get() = bot.getGroup(id)!!
}